example-app/src/main/java/com/adyen/checkout/example/ui/blik/BlikActivity.kt

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by onurk on 17/11/2022.
+ */
+
+package com.adyen.checkout.example.ui.blik
+
+import android.content.Intent
+import android.os.Bundle
+import android.widget.Toast
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.isVisible
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import com.adyen.checkout.await.AwaitComponent
+import com.adyen.checkout.blik.BlikComponent
+import com.adyen.checkout.components.ActionComponentEvent
+import com.adyen.checkout.components.model.paymentmethods.PaymentMethod
+import com.adyen.checkout.example.R
+import com.adyen.checkout.example.databinding.ActivityBlikBinding
+import com.adyen.checkout.example.ui.configuration.CheckoutConfigurationProvider
+import com.adyen.checkout.redirect.RedirectComponent
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class BlikActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityBlikBinding
+    private val blikViewModel: BlikViewModel by viewModels()
+
+    @Inject
+    internal lateinit var checkoutConfigurationProvider: CheckoutConfigurationProvider
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        // Insert return url in extras, so we can access it in the ViewModel through SavedStateHandle
+        val returnUrl = RedirectComponent.getReturnUrl(applicationContext) + "/blik"
+        intent = (intent ?: Intent()).putExtra(RETURN_URL_EXTRA, returnUrl)
+
+        binding = ActivityBlikBinding.inflate(layoutInflater)
+        binding.payButton.setOnClickListener { blikViewModel.onPayClick() }
+        setContentView(binding.root)
+
+        setSupportActionBar(binding.toolbar)
+        supportActionBar?.setDisplayShowTitleEnabled(true)
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        supportActionBar?.setTitle(R.string.blik_title)
+
+        binding.toolbar.setNavigationOnClickListener {
+            finish()
```

Comment:
It's better to call `onBackPressed()`. Just in case there is would be more complex navigation